File: fm-domain/src/main/java/io/zeitwert/fm/building/adapter/api/jsonapi/impl/ObjBuildingApiRepositoryImpl.java
package io.zeitwert.fm.building.adapter.api.jsonapi.impl;

import io.zeitwert.dddrive.app.model.SessionContext;
import io.zeitwert.dddrive.ddd.adapter.api.jsonapi.base.AggregateApiRepositoryBase;
import io.zeitwert.fm.building.adapter.api.jsonapi.ObjBuildingApiRepository;
import io.zeitwert.fm.building.adapter.api.jsonapi.dto.ObjBuildingDto;
import io.zeitwert.fm.building.model.ObjBuilding;
import io.zeitwert.fm.building.model.ObjBuildingRepository;
import io.zeitwert.fm.oe.model.ObjUserRepository;
import org.springframework.stereotype.Controller;

@Controller("objBuildingApiRepository")
public class ObjBuildingApiRepositoryImpl
		extends AggregateApiRepositoryBase<ObjBuilding, ObjBuildingDto>
		implements ObjBuildingApiRepository {

	public ObjBuildingApiRepositoryImpl(
			ObjBuildingRepository repository,
			SessionContext requestCtx,
			ObjUserRepository userRepository,
			ObjBuildingDtoAdapter dtoAdapter) {
		super(ObjBuildingDto.class, requestCtx, userRepository, repository, dtoAdapter);
	}

}

File: fm-domain/src/main/java/io/zeitwert/fm/building/adapter/api/jsonapi/impl/ObjBuildingApiRepositoryImpl.kt
package io.zeitwert.fm.building.adapter.api.jsonapi.impl

import dddrive.ddd.core.model.RepositoryDirectory
import io.zeitwert.dddrive.app.model.SessionContext
import io.zeitwert.dddrive.ddd.adapter.api.jsonapi.base.GenericAggregateApiRepositoryBase
import io.zeitwert.fm.building.adapter.api.jsonapi.dto.ObjBuildingDto
import io.zeitwert.fm.building.model.ObjBuilding
import io.zeitwert.fm.building.model.ObjBuildingRepository
import org.springframework.stereotype.Controller

@Controller("objBuildingApiRepository")
open class ObjBuildingApiRepositoryImpl(
	directory: RepositoryDirectory,
	repository: ObjBuildingRepository,
	adapter: ObjBuildingDtoAdapter,
	sessionCtx: SessionContext,
) : GenericAggregateApiRepositoryBase<ObjBuilding, ObjBuildingDto>(
		resourceClass = ObjBuildingDto::class.java,
		directory = directory,
		repository = repository,
		adapter = adapter,
		sessionCtx = sessionCtx,
	)
